### DISCUSSION TITLE
Re-run single invocation of parametrised test in Junit 5

### DISCUSSION THREAD

**Discussion by AlexShmydov** [allure-framework/allure2#2522] [Questions & Support] [answered]
Hi all!

As i see, io.qameta.allure.junitplatform.AllureJunitPlatform has the next code (https://github.com/allure-framework/allure-java/blob/main/allure-junit-platform/src/main/java/io/qameta/allure/junitplatform/AllureJunitPlatform.java#L578) when test is starting:

```
        testSource.flatMap(AllureJunitPlatformUtils::getFullName).ifPresent(result::setFullName);
```

The AllureJunitPlatformUtils class (https://github.com/allure-framework/allure-java/blob/main/allure-junit-platform/src/main/java/io/qameta/allure/junitplatform/AllureJunitPlatformUtils.java#L41):

```
 if (source instanceof MethodSource) {
            final MethodSource ms = (MethodSource) source;
            return Optional.of(String.format("%s.%s", ms.getClassName(), ms.getMethodName()));
        }
```

will return just class name + method name, without any arguments.

But it is not a unique name for parametrised tests. 
The fullName will be the same for all invocations of parametrised tests. And all invocations relate to the one test case (see screnshot)

![image](https://github.com/allure-framework/allure2/assets/32179689/4761f2ab-c0ce-46f7-8fb0-2a86144dbf34)

When we try to rerun not all invocations of parametrised tests - it will run whole test, cause the TestOps sends a test plan with the same fullName values. Example:

```
{
  "version": "1.0",
  "tests": [
    {
      "id": 51010,
      "selector": "com.example.test.SuperTest.checkThatIsTrue"
    },
    {
      "id": 51010,
      "selector": "com.example.test.SuperTest.checkThatIsTrue"
    }
  ]
}
```

Than, the AllurePostDiscoveryFilter tries to filter tests via the next parameters (https://github.com/allure-framework/allure-java/blob/main/allure-junit-platform/src/main/java/io/qameta/allure/junitplatform/AllurePostDiscoveryFilter.java#L63) :
```
final String allureId = findAllureId(object);
final String uniqueId = object.getUniqueId().toString();
 final String fullName = object.getSource()
                .flatMap(AllureJunitPlatformUtils::getFullName)
                .orElse(null);
```

UniqueId (which provides from junit5) will looks like: `[engine:junit-jupiter]/[class:com.example.test.SuperTest]/[test-template:checkThatIsTrue(java.lang.Integer)]` for all invocations. 

So, all invocations will be re-run.

**Is it possible to re-run single invocation of parametrised tests?** 

I know, that is possible in cucumber (which based on Junit) and it is possible via IDEA (https://github.com/junit-team/junit5/discussions/3794)

P.S. 
UniqueId for a single invocation should have the next value, for example:
```
[engine:junit-jupiter]/[class:com.example.test.SuperTest]/[test-template:checkThatIsTrue(java.lang.Integer)]/[test-template-invocation:#3]
```

P.S.S Also, I can not add AllureId annotation above each argument in parametrised tests to filter tests by allureId

**Comment by baev** [comment, marked as answer]
We have no plans to add support for retiring tests with only a specific set of parameters. Our primary focus is to support selective runs for all the Allure integrations. 

Having support for reruns with a specific set of parameters will dramatically increase the complexity level.
